pandas/core/accessor.py

```diff
@@ -129,3 +134,109 @@ def f(self, *args, **kwargs):
             # don't overwrite existing methods/properties
             if overwrite or not hasattr(cls, name):
                 setattr(cls, name, f)
+
+
+# Ported with modifications from xarray
+# https://github.com/pydata/xarray/blob/master/xarray/core/extensions.py
```

Comment:
Indeed, please do let me know exactly what you needed to change :)

Reply:
> Here's the relevant diff between xarray and pandas:

@TomAugspurger : For better or worse, that's a much less significant change than I expected.  Maybe should add a comment about that.